src/routes/InstancesPage/InstancesToolbarSearchFilter.js

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ToolbarToggleGroup,
+  ToolbarGroup,
+  ToolbarFilter,
+  SelectVariant,
+  SelectOption,
+  Select,
+  ToolbarItem,
+  InputGroup,
+  TextInput,
+  Button,
+} from '@patternfly/react-core';
+import { FilterIcon, SearchIcon } from '@patternfly/react-icons';
+import SelectSingle from '../../components/SelectSingle';
+
+function InstancesToolbarSearchFilter({ filters, setFilters }) {
+  const [selectedFilter, setSelectedFilter] = useState('Name');
+  // local state for input values
+  const [inputName, setInputName] = useState('');
+  const [inputOwner, setInputOwner] = useState('');
+  // local state for Select isExpanded values
+  // @TODO: We can refactor the SelectSingle component to be more reusable for the usecase in this component as well. Then we don't need to keep this state here.
+  const [isRegionExpanded, setIsRegionExpanded] = useState(false);
+  const [isStatusExpanded, setIsStatusExpanded] = useState(false);
+
+  // TODO: Extract into separate utils file to be reused in other cases
```

Comment:
Added TODOs for improvements. If we need to use Toolbar filters in other places, we'll eventually need to extract some functions out into a utils file to be more reusable